zfsbootmenu/pre-init/zfsbootmenu-parse-commandline.sh

```diff
@@ -219,30 +219,40 @@ if kcl_override=$( get_zbm_arg zbm.kcl_override ) ; then
   zinfo "overriding all BE KCLs with: '$( kcl_assemble < "${BASE}/cmdline" )'"
 fi
 
-zbm_prefer_pool=
-if zbm_prefer_pool=$( get_zbm_arg zbm.prefer ) ; then
-  # shellcheck disable=SC2034
-  zbm_prefer_pool="${zbm_prefer_pool%%/*}"
-  zinfo "preferring ${zbm_prefer_pool} for bootfs"
-fi
-
 zbm_wait_for_devices=
 if zbm_wait_for_devices=$( get_zbm_arg zbm.wait_for ) ; then
   zinfo "system will wait for ${zbm_wait_for_devices}"
 fi
 
-# pool! : this pool must be imported before all others
-# pool!!: this pool, and no others, must be imported
+zbm_prefer_bootfs=
+if zbm_prefer_bootfs=$( get_zbm_arg zbm.bootfs ) ; then
+  zinfo "using ${zbm_prefer_bootfs} for bootfs"
+fi
+
+zbm_prefer_pool=
+if zbm_prefer_pool=$( get_zbm_arg zbm.prefer ) ; then
+  # shellcheck disable=SC2034
+  zbm_prefer_pool="${zbm_prefer_pool%%/*}"
+  [ -z  "${zbm_prefer_bootfs}" ] && zinfo "preferring ${zbm_prefer_pool} for bootfs"
+fi
 
 # shellcheck disable=SC2034
 case "${zbm_prefer_pool}" in
   *!!)
     zbm_require_pool="only"
     zbm_prefer_pool="${zbm_prefer_pool%!!}"
+    zinfo "will only attempt to import ${zbm_prefer_pool}"
+
+    if [ -n "${zbm_prefer_bootfs}" ] && [ "${zbm_prefer_pool}" != "${zbm_prefer_bootfs%%/*}" ]; then
+      zerror "zbm.bootfs '${zbm_prefer_bootfs}' not on the exclusive pool import"
+      unset zbm_prefer_bootfs
+    fi
+
```

Comment:
I wouldn't resolve this here and, more generally, wouldn't treat a missing `zbm.bootfs` as something to ignore. If somebody specifies a non-existent or improper filesystem for `zbm.bootfs`, the intent is unclear. This should trigger a drop to the menu rather than moving on to reading the `bootfs` pool property.

I think removing this block and relying on your existing changes in `50-import-pools` will achieve this effect.

Reply:
no longer needed now that `zbm.prefer` has been extended